File: datasets_v2/finance/gnucash/compose/activity_transaction_detail.kt
import androidx.compose.ui.tooling.preview.Preview
import androidx.compose.foundation.background
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.lazy.items
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.paint
import androidx.compose.ui.graphics.Color
import androidx.compose.foundation.Image
import androidx.compose.ui.res.painterResource
import androidx.compose.ui.layout.ContentScale
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.sp
import androidx.compose.ui.unit.dp

@Composable
fun activity_transaction_detail() {
  Column(modifier = Modifier.fillMaxWidth().fillMaxHeight()) {
    Column(modifier = Modifier.fillMaxWidth().fillMaxHeight()) {
      Column(modifier = Modifier.background(Color.Unspecified).fillMaxWidth().height(128.0.dp)) {
        Column(modifier = Modifier.fillMaxWidth().wrapContentHeight()) {
          Text(text = "Transaction description which can be rather long", modifier = Modifier.fillMaxWidth().wrapContentHeight().padding(bottom = 20.0.dp))
          Text(text = "Expenses:Computer", modifier = Modifier.fillMaxWidth().wrapContentHeight().padding(start = 8.0.dp), color = Color.White)
        }
      }
      Column(modifier = Modifier.fillMaxWidth().fillMaxHeight().verticalScroll(rememberScrollState())) {
        Column(modifier = Modifier.fillMaxWidth().wrapContentHeight().padding(10.0.dp).padding(top = 20.0.dp)) {
          Row(modifier = Modifier.fillMaxWidth().wrapContentHeight().padding(top = 5.0.dp).padding(bottom = 5.0.dp).padding(start = 6.0.dp), verticalAlignment = Alignment.CenterVertically) {
            Text(text = "Balance", modifier = Modifier.wrapContentHeight().weight(2.0f), color = Color.Black, fontSize = 16.sp, textAlign = TextAlign.Center)
            Text(text = "$ 2000", modifier = Modifier.wrapContentHeight().weight(1.5f), color = Color.Black, fontSize = 16.sp, textAlign = TextAlign.End)
            Text(text = "$ 2000", modifier = Modifier.wrapContentHeight().weight(1.5f), color = Color.Black, fontSize = 16.sp, textAlign = TextAlign.End)
          }
          Row(modifier = Modifier.fillMaxWidth().padding(top = 20.0.dp).padding(bottom = 10.0.dp)) {
            Image(painter = painterResource(id = R.drawable.ic_action_time), contentDescription = null, modifier = Modifier.wrapContentHeight().weight(0.5f).padding(end = 10.0.dp), contentScale = ContentScale.Fit)
            Text(text = "", modifier = Modifier.wrapContentHeight().weight(5.0f), color = Color.Black, fontSize = 16.sp)
          }
          Row(modifier = Modifier.fillMaxWidth().padding(top = 10.0.dp).padding(bottom = 10.0.dp)) {
            Image(painter = painterResource(id = R.drawable.ic_action_sort_by_size), contentDescription = null, modifier = Modifier.wrapContentHeight().weight(0.5f).padding(end = 10.0.dp), contentScale = ContentScale.Fit)
            Text(text = "", modifier = Modifier.wrapContentHeight().weight(5.0f), color = Color.Black, fontSize = 16.sp)
          }
          Row(modifier = Modifier.fillMaxWidth().padding(top = 10.0.dp).padding(bottom = 10.0.dp)) {
            Image(painter = painterResource(id = R.drawable.ic_action_rotate_right), contentDescription = null, modifier = Modifier.wrapContentHeight().weight(0.5f).padding(end = 10.0.dp), contentScale = ContentScale.Fit)
            Text(text = "", modifier = Modifier.wrapContentHeight().weight(5.0f), color = Color.Black, fontSize = 16.sp)
          }
        }
      }
    }
    Column(modifier = Modifier.size(40.0.dp, 40.0.dp).padding(start = 6.0.dp)) {
    }
  }
}


@Preview(showBackground = true)
@Composable
fun activity_transaction_detail_preview() {
    MyApplicationTheme {
        activity_transaction_detail()
    }
}

File: experiments/finance/gnucash/swiftui/Activity_transaction_detail.swift
import SwiftUI

struct Activity_transaction_detail: View {
  var body: some View {
    VStack(alignment: .leading, spacing: 0) {
      VStack(alignment: .leading, spacing: 0) {
        VStack(alignment: .leading, spacing: 0) {
          VStack(alignment: .leading, spacing: 0) {
            Text("Transaction description which can be rather long").frame(maxWidth: .infinity, alignment: .leading).padding(.bottom, 20.0)
            Text("Expenses:Computer").foregroundColor(Color.white).frame(maxWidth: .infinity, alignment: .leading).padding(.leading, 8.0)
          }
          .frame(maxWidth: .infinity)
        }
        .background(Color.clear)
        .frame(maxWidth: .infinity)
        .frame(height: 128.0)
        ScrollView {
          VStack(alignment: .leading, spacing: 0) {
            HStack(alignment: .center, spacing: 0) {
              Text("Balance").font(.system(size: 16)).foregroundColor(Color.black).multilineTextAlignment(.center).frame(maxWidth: .infinity, alignment: .center).frame(maxWidth: .infinity)
              Text("$ 2000").font(.system(size: 16)).foregroundColor(Color.black).multilineTextAlignment(.trailing).frame(maxWidth: .infinity, alignment: .trailing).frame(maxWidth: .infinity).frame(maxHeight: .infinity, alignment: .center)
              Text("$ 2000").font(.system(size: 16)).foregroundColor(Color.black).multilineTextAlignment(.trailing).frame(maxWidth: .infinity, alignment: .trailing).frame(maxWidth: .infinity).frame(maxHeight: .infinity, alignment: .center)
            }
            .frame(maxWidth: .infinity)
            .padding(.top, 5.0)
            .padding(.bottom, 5.0)
            .padding(.leading, 6.0)
            HStack(alignment: .center, spacing: 0) {
              Image("ic_action_time").resizable().scaledToFit().frame(maxWidth: .infinity).padding(.trailing, 10.0)
              Text("").font(.system(size: 16)).foregroundColor(Color.black).frame(maxWidth: .infinity, alignment: .leading).frame(maxWidth: .infinity)
            }
            .frame(maxWidth: .infinity)
            .padding(.top, 20.0)
            .padding(.bottom, 10.0)
            HStack(alignment: .center, spacing: 0) {
              Image("ic_action_sort_by_size").resizable().scaledToFit().frame(maxWidth: .infinity).padding(.trailing, 10.0)
              Text("").font(.system(size: 16)).foregroundColor(Color.black).frame(maxWidth: .infinity, alignment: .leading).frame(maxWidth: .infinity)
            }
            .frame(maxWidth: .infinity)
            .padding(.top, 10.0)
            .padding(.bottom, 10.0)
            HStack(alignment: .center, spacing: 0) {
              Image("ic_action_rotate_right").resizable().scaledToFit().frame(maxWidth: .infinity).padding(.trailing, 10.0)
              Text("").font(.system(size: 16)).foregroundColor(Color.black).frame(maxWidth: .infinity, alignment: .leading).frame(maxWidth: .infinity)
            }
            .frame(maxWidth: .infinity)
            .padding(.top, 10.0)
            .padding(.bottom, 10.0)
          }
          .frame(maxWidth: .infinity)
          .padding(10.0)
          .padding(.top, 20.0)
        }
        .frame(maxWidth: .infinity, maxHeight: .infinity)
      }
      .frame(maxWidth: .infinity, maxHeight: .infinity)
      VStack(alignment: .leading, spacing: 0) {
      }
      .frame(width: 40.0, height: 40.0)
      .padding(.leading, 6.0)
    }
    .frame(maxWidth: .infinity, maxHeight: .infinity)
  }
}

struct Activity_transaction_detail_Previews: PreviewProvider {
  static var previews: some View {
    Activity_transaction_detail()
  }
}
